src/main/java/com/coreos/jetcd/EtcdLease.java

```diff
@@ -0,0 +1,98 @@
+package com.coreos.jetcd;
+
+import com.coreos.jetcd.api.LeaseGrantResponse;
+import com.coreos.jetcd.api.LeaseKeepAliveResponse;
+import com.coreos.jetcd.api.LeaseRevokeResponse;
+import com.google.common.util.concurrent.ListenableFuture;
+
+import java.security.InvalidParameterException;
+
+/**
+ * Interface of Lease talking to etcd
+ */
+public interface EtcdLease {
+
+    /**
+     * New a lease with ttl value
+     *
+     * @param ttl ttl value, unit seconds
+     * @return
+     */
+    ListenableFuture<LeaseGrantResponse> grant(long ttl);
+
+    /**
+     * revoke one lease and the key bind to this lease will be removed
+     *
+     * @param leaseId id of the lease to revoke
+     * @return
+     */
+    ListenableFuture<LeaseRevokeResponse> revoke(long leaseId);
+
+    /**
+     * keep alive one lease in background
+     *
+     * @param leaseId          id of lease to set handler
+     * @param etcdLeaseHandler the handler for the lease, this value can be null
+     */
+    void keepAlive(long leaseId, EtcdLeaseHandler etcdLeaseHandler);
+
+    /**
+     * keep alive one lease only once
+     *
+     * @param leaseId id of lease to keep alive once
+     * @return The keep alive response
+     */
+    ListenableFuture<LeaseKeepAliveResponse> keepAliveOnce(long leaseId);
+
+    /**
+     * set EtcdLeaseHandler for lease
+     *
+     * @param leaseId          id of the lease to set handler
+     * @param etcdLeaseHandler the handler for the lease
+     * @throws InvalidParameterException if lease do not exist
+     */
+    void setEtcdLeaseHandler(long leaseId, EtcdLeaseHandler etcdLeaseHandler) throws InvalidParameterException;
```

Comment:
`InvalidParameterException` should be changed to other exception, if can not find one in Java, you can define custom one.